drivers/sensor/icm42605/icm42605.c

```diff
@@ -0,0 +1,392 @@
+static void icm42605_convert_gyro(struct sensor_value *val, int16_t raw_val,
+				  uint16_t sensitivity_x10)
+{
+	int64_t conv_val;
+
+	conv_val = ((int64_t)raw_val * SENSOR_PI * 10) /
+		   (sensitivity_x10 * 180U);
+	val->val1 = conv_val / 1000000;
+	val->val2 = conv_val % 1000000;
+}
+
+/* see "Temperature Measurement" section from register map description */
+static inline void icm42605_convert_temp(struct sensor_value *val,
+					 int16_t raw_val)
+{
+	val->val1 = (((int64_t)raw_val * 100) / 207) + 25;
+	val->val2 = ((((int64_t)raw_val * 100) % 207) * 1000000) / 207;
+
+	if (val->val2 < 0) {
+		val->val1--;
+		val->val2 += 1000000;
+	} else if (val->val2 >= 1000000) {
+		val->val1++;
+		val->val2 -= 1000000;
+	}
+}
+
+static int icm42605_channel_get(struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct icm42605_data *drv_data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		icm42605_convert_accel(val, drv_data->accel_x,
+				       drv_data->accel_sensitivity_shift);
+		icm42605_convert_accel(val + 1, drv_data->accel_y,
+				       drv_data->accel_sensitivity_shift);
+		icm42605_convert_accel(val + 2, drv_data->accel_z,
+				       drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_X:
+		icm42605_convert_accel(val, drv_data->accel_x,
+				       drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		icm42605_convert_accel(val, drv_data->accel_y,
+				       drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		icm42605_convert_accel(val, drv_data->accel_z,
+				       drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_GYRO_XYZ:
+		icm42605_convert_gyro(val, drv_data->gyro_x,
+				      drv_data->gyro_sensitivity_x10);
+		icm42605_convert_gyro(val + 1, drv_data->gyro_y,
+				      drv_data->gyro_sensitivity_x10);
+		icm42605_convert_gyro(val + 2, drv_data->gyro_z,
+				      drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_X:
+		icm42605_convert_gyro(val, drv_data->gyro_x,
+				      drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		icm42605_convert_gyro(val, drv_data->gyro_y,
+				      drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		icm42605_convert_gyro(val, drv_data->gyro_z,
+				      drv_data->gyro_sensitivity_x10);
+		break;
+	default: /* chan == SENSOR_CHAN_DIE_TEMP */
+		icm42605_convert_temp(val, drv_data->temp);
+	}
+
+	return 0;
+}
+
+static int icm42605_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	int result = 0;
+	uint16_t fifo_count = 0;
+	static uint8_t data[HARDWARE_FIFO_SIZE] = { 0, };
+	struct icm42605_data *drv_data = dev->data;
+
+	if (drv_data->tap_en) {
+		result = inv_spi_read(REG_INT_STATUS3, data, 1);
+
+		if (data[0] & BIT_INT_STATUS_TAP_DET) {
+			LOG_DBG("Tap detected!!!");
+			result = inv_spi_read(REG_APEX_DATA4, data, 1);
+			if (data[0] & 0x08) {
```

Comment:
constants should be defined. Another question here, what does driver do in case of tap detection, only logging?

Reply:
I misses value set of tap and double tap, I will implement that feature.

Reply:
Implemented trigger of tap/double tap.
Sample application will be commit at next time.